hotel/src/hil/common.rs

```diff
@@ -0,0 +1,25 @@
+/// Possible errors returned by syscalls. In case of failure, the negative value of the error is
+/// returned to the application.
+#[derive(Copy, Clone)]
+pub enum SyscallError {
```

Comment:
FWIW, I don't (personally) think it's worth blocking progress (e.g. this pull request) on this discussion. Should be fairly straight forward to go back and make changes after the fact.

Meanwhile, no longer my repo (it has transferred to @domrizzo's). Do you want feedback from me? Or waiting for Dom/others?